libs/pika/async_mpi/tests/unit/CMakeLists.txt

```
@@ -37,18 +54,27 @@ foreach(test ${tests})
     FOLDER "Tests/Unit/Modules/AsyncMPI"
   )
 
-  foreach(polling_mode RANGE 0 ${PIKA_MPI_MODES_LOOP_COUNT} 1)
-    pika_add_pseudo_target(${test}_mode_${polling_mode}_test)
-    pika_add_pseudo_dependencies(${test}_mode_${polling_mode}_test ${test}_test)
-    pika_add_unit_test(
-      "modules.async_mpi"
-      ${test}_mode_${polling_mode}
-      ${${test}_PARAMETERS}
-      EXECUTABLE
-      ${test}_test
-      ARGS
-      "--pika:mpi-completion-mode=${polling_mode}"
-    )
+  # if NO_POOL is set, then we only use enable_pool=false and skip true
+  set(BOOL_RANGE 1) # 0..1
+  if(${${test}_NO_POOL})
+    set(BOOL_RANGE 0) # just 0
+  endif()
+
+  foreach(enable_pool RANGE ${BOOL_RANGE})
+    foreach(polling_mode RANGE 0 ${PIKA_MPI_MODES_LOOP_COUNT} 1)
+      set(full_name_ ${test}_mode_${enable_pool}_${polling_mode})
+      pika_add_pseudo_target(${full_name_}_test)
+      pika_add_pseudo_dependencies(${full_name_}_test ${test}_test)
+      pika_add_unit_test(
+        "modules.async_mpi"
+        ${full_name_}
+        ${${test}_PARAMETERS}
+        EXECUTABLE
+        ${test}
```

This should still have the `_test` suffix. The bug is in `pika_add_test`, not here.
```suggestion
        ${test}_test
```